pkg/resourcemanager/controller/garbagecollector/references/references.go

```diff
@@ -144,6 +145,16 @@ func InjectAnnotations(obj runtime.Object, additional ...string) error {
 	case *resourcesv1alpha1.ManagedResource:
 		referenceAnnotations := computeAnnotationsFromLocalObjRefs(o.Spec.SecretRefs, KindSecret, additional...)
 		o.Annotations = mergeAnnotations(o.Annotations, referenceAnnotations)
+	case *monitoringv1.Prometheus:
+		var references []corev1.LocalObjectReference
+		for _, remoteWrite := range o.Spec.RemoteWrite {
+			if basicAuth := remoteWrite.BasicAuth; basicAuth != nil {
+				references = append(references, basicAuth.Username.LocalObjectReference)
+				references = append(references, basicAuth.Password.LocalObjectReference)
+			}
+		}
+		referenceAnnotations := computeAnnotationsFromLocalObjRefs(references, KindSecret, additional...)
+		o.Annotations = mergeAnnotations(o.Annotations, referenceAnnotations)
```

Comment:
We also need to check additionally injected containers/volumes (like the cortex): https://github.com/gardener/gardener/blob/490a03c4185ff017d37d436f4e02626e6c4eec71/pkg/component/observability/monitoring/prometheus/prometheus.go#L158-L159